app/controllers/RiderController.java

```diff
@@ -19,14 +27,101 @@ public RiderController(RiderRepository riderRepository) {
         this.riderRepository = riderRepository;
     }
 
-    public Result index() {
-        return ok("RIDER REPO");
+    public CompletionStage<Result> getRiders() {
+        return riderRepository.getAllRiders().thenApplyAsync(riders -> {
+            return ok(riders);
+        }).exceptionally(ex -> {
+            Result res = null;
+            switch (ExceptionUtils.getRootCause(ex).getClass().getSimpleName()){
+                case "IndexOutOfBoundsException":
+                    res = badRequest("No riders are set in DB.");
+                    break;
+                default:
+                    res = internalServerError(ex.getMessage());
+            }
+            return res;
+        });
     }
 
-    public CompletionStage<Result> getRiders() {
-        /*return riderRepository.list().thenApplyAsync(riderStream -> {
-           return ok(toJson(riderStream.collect(Collectors.toList())));
-        });*/
-        return null;
+    public CompletionStage<Result> getRider(int riderId){
+        return riderRepository.getRider(riderId).thenApplyAsync(rider -> {
+            return ok(rider);
+        }).exceptionally(ex -> {
+            Result res = null;
+            switch (ExceptionUtils.getRootCause(ex).getClass().getSimpleName()){
+                case "NoResultException":
+                    res = badRequest("No rider with id: " + riderId + " is available in DB.");
+                    break;
+                default:
+                    res = internalServerError(ex.getMessage());
+            }
+            return res;
+        });
+    }
+
+    @BodyParser.Of(BodyParser.Json.class)
+    public CompletionStage<Result> addRider(){
+        JsonNode json = request().body().asJson();
+        return parseRider(json).thenApply(rider -> riderRepository.addRider(rider)).thenApply(rider -> {
+            return ok(rider + " has been added");
+        }).exceptionally(ex -> {
+            Result res = null;
+            switch (ExceptionUtils.getRootCause(ex).getClass().getSimpleName()){
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this "switch" statement by "if" statements to increase readability. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1301)